compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -206,8 +207,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     })
                 }
                 ty::Param(param) => {
-                    let def_id = self.tcx.generics_of(self.body_id.owner).type_param(&param, self.tcx).def_id;
-                    self.tcx.predicates_of(self.body_id.owner).predicates.iter().find_map(|(pred, _)| {
+                    let body_id = self.tcx.hir().local_def_id_to_hir_id(self.body_id);
+                    let def_id = self.tcx.generics_of(body_id.owner).type_param(&param, self.tcx).def_id;
+                    self.tcx.predicates_of(body_id.owner).predicates.iter().find_map(|(pred, _)| {
```

Comment:
again, you can directly use `self.body_id` here instead of using the `HirId` owner